File: frontend/src/types/operation-log.ts
// 操作日志类型定义

export interface OperationLog {
  id?: string;
  module: string;          // 模块名称：样衣开发、下单管理等
  operation: string;       // 操作类型：删除、新增、修改等
  operatorId: string;      // 操作人ID
  operatorName: string;    // 操作人姓名
  targetType: string;      // 目标类型：款式、订单、物料等
  targetId: string;        // 目标ID
  targetName?: string;     // 目标名称：款号、订单号等
  reason?: string;         // 操作原因
  details?: string;        // 详细信息（JSON格式）
  ip?: string;            // 操作IP
  userAgent?: string;     // 浏览器信息
  operationTime: string;  // 操作时间
  status: 'success' | 'failure';  // 操作状态
  errorMessage?: string;  // 错误信息（失败时）
}

export interface OperationLogQueryParams {
  module?: string;         // 模块筛选
  operation?: string;      // 操作类型筛选
  operatorName?: string;   // 操作人筛选
  targetType?: string;     // 目标类型筛选
  startDate?: string;      // 开始日期
  endDate?: string;        // 结束日期
  page: number;
  pageSize: number;
}

File: backend/src/main/java/com/fashion/supplychain/system/entity/OperationLog.java
package com.fashion.supplychain.system.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import com.baomidou.mybatisplus.annotation.FieldFill;
import com.baomidou.mybatisplus.annotation.TableField;
import lombok.Data;

import java.time.LocalDateTime;

/**
 * 操作日志实体
 */
@Data
@TableName("t_operation_log")
public class OperationLog {

    @TableId(type = IdType.AUTO)
    private Long id;

    /**
     * 模块名称
     */
    private String module;

    /**
     * 操作类型
     */
    private String operation;

    /**
     * 操作人ID
     */
    private Long operatorId;

    /**
     * 操作人姓名
     */
    private String operatorName;

    /**
     * 目标类型
     */
    private String targetType;

    /**
     * 目标ID
     */
    private String targetId;

    /**
     * 目标名称
     */
    private String targetName;

    /**
     * 操作原因
     */
    private String reason;

    /**
     * 详细信息（JSON格式）
     */
    private String details;

    /**
     * 人类可读的变更摘要（如"商品名称：绿色披肩->松绿云缂罗锦披肩；款式编码：BR26CA0050A->BR26CA0050B"）
     */
    private String changeSummary;

    /**
     * 操作IP
     */
    private String ip;

    /**
     * 浏览器信息
     */
    private String userAgent;

    /**
     * 操作时间
     */
    private LocalDateTime operationTime;

    /**
     * 状态：success/failure
     */
    private String status;

    /**
     * 错误信息
     */
    private String errorMessage;

    @TableField(fill = FieldFill.INSERT)
    private Long tenantId;
}
